QPong-Python-Server/flask_server/api.py

```diff
@@ -1,22 +1,62 @@
 #!/usr/bin/env python3
 
-from qiskit import QuantumRegister, ClassicalRegister
-from qiskit import QuantumCircuit, Aer, execute, IBMQ
-
-
-def run_qasm(qasm, backend_to_run="qasm_simulator", api_token=None, shots=1024, memory=False):
-    if api_token:
-        IBMQ.enable_account(api_token, 'https://api.quantum-computing.ibm.com/api/Hubs/ibm-q/Groups/open/Projects/main')
-    qc = QuantumCircuit.from_qasm_str(qasm)
-    backend = Aer.get_backend(backend_to_run)
-    job_sim = execute(qc, backend, shots=shots, memory=memory)
-    sim_result = job_sim.result()
-    if memory:
-        return sim_result.get_memory(qc)
-    else:
-        return sim_result.get_counts(qc)
-
-
-def backend_configuration(backend_to_run="qasm_simulator"):
-    backend = Aer.get_backend(backend_to_run)
-    return backend.configuration().as_dict()
\ No newline at end of file
+from qiskit import BasicAer, execute, ClassicalRegister
+from copy import deepcopy
+import json_tricks
+
+from model.circuit_grid_model import CircuitGridModel, CircuitGridNode
+from model import circuit_node_types as node_types
+
+QUBIT_NUM = 3
+
+
+def statevector(gate_array_string):
+    circuit = circuit_from_string(gate_array_string)
+    shot_num = 1000
+
+    backend_sv_sim = BasicAer.get_backend('statevector_simulator')
+    job_sim = execute(circuit, backend_sv_sim, shots=shot_num)
+    result_sim = job_sim.result()
+    quantum_state = result_sim.get_statevector(circuit, decimals=3)
+
+    return json_tricks.dumps(quantum_state)
+
+
+def measurement(gate_array_string):
+    circuit = circuit_from_string(gate_array_string)
+    shot_num = 1
+
+    backend_sv_sim = BasicAer.get_backend('qasm_simulator')
+    cr = ClassicalRegister(QUBIT_NUM)
+    measure_circuit = deepcopy(circuit)  # make a copy of circuit
+    measure_circuit.add_register(cr)  # add classical registers for measurement readout
+    measure_circuit.measure(measure_circuit.qregs[0], measure_circuit.cregs[0])
+    job_sim = execute(measure_circuit, backend_sv_sim, shots=shot_num)
+    result_sim = job_sim.result()
+    counts = result_sim.get_counts(circuit)
+
+    state_in_decimal = int(list(counts.keys())[0], 2)
+
+    return str(state_in_decimal)
+
+
+def circuit_from_string(gate_array_string):
+    gate_array = gate_array_string.split(',')
+    row_max = 3
```

Comment:
shouldn't this be set by the constant `QUBIT_NUM`?

Reply:
Yes, I will change that